File: acmicpc/acmicpc01179.py
import sys

sys.stdin = open('acmicpc/sample.txt', 'r')

v, e = map(int, sys.stdin.readline().split())

weights = []
for i in range(e):
    node, nxt, w = map(int, sys.stdin.readline().split())
    weights.append([w, node, nxt])

weights.sort()
parent = [i for i in range(v+1)]

def find(x):
    if parent[x] == x:
        return x
    else:
        return find(parent[x])

answer = 0
for w, node, nxt in weights:
    x = find(node)
    y = find(nxt)
    if x != y:
        parent[y] = x
        answer += w

print(answer)

File: acmicpc/acmicpc01179.cpp
#include <iostream>
#include <algorithm>
#include <vector>

using namespace std;

vector<int> parent;

int find(int x){
    if (parent[x] == x){
        return x;
    } else {
        return find(parent[x]);
    }
}


int main(){
    // freopen("sample.txt", "r", stdin);

    int v, e;
    cin >> v >> e;
    vector< vector<int> > graph;

    parent.push_back(0);
    for (int i=1; i < (v + 1); i++){
        parent.push_back(i);
    }

    for (int i=0; i < e; i++){
        vector<int> temp;
        int node, nxt, w;
        cin >> node >> nxt >> w;
        temp.push_back(w);
        temp.push_back(node);
        temp.push_back(nxt);
        graph.push_back(temp);
    }
    
    sort(graph.begin(), graph.end());

    int answer = 0;
    for (int i=0; i < e; i++){
        int x, y;
        x = find(graph[i][1]);
        y = find(graph[i][2]);
        if (x != y){
            parent[y] = x;
            answer = answer + graph[i][0];
        }
    }
    
    cout << answer << "\n";
    
}